src/librbd/io/ReadResult.cc

```diff
@@ -106,15 +103,21 @@ void ReadResult::C_ImageReadRequest::finish(int r) {
     r = length;
   }
 
-  C_ReadRequest::finish(r);
+  aio_completion->complete_request(r);
 }
 
-void ReadResult::C_SparseReadRequestBase::finish(ExtentMap &extent_map,
-                                                 const Extents &buffer_extents,
-                                                 uint64_t offset, size_t length,
-                                                 bufferlist &bl, int r) {
+ReadResult::C_ObjectReadRequest::C_ObjectReadRequest(
+    AioCompletion *aio_completion, uint64_t object_off, uint64_t object_len,
+    Extents&& buffer_extents, bool ignore_enoent)
+  : aio_completion(aio_completion), object_off(object_off),
+    object_len(object_len),buffer_extents(std::move(buffer_extents)),
+    ignore_enoent(ignore_enoent) {
+  aio_completion->add_request();
+}
+
+void ReadResult::C_ObjectReadRequest::finish(int r) {
   CephContext *cct = aio_completion->ictx->cct;
-  ldout(cct, 10) << "C_SparseReadRequestBase: r = " << r
+  ldout(cct, 10) << "C_ObjectReadRequest: r = " << r
```

Comment:
nit: while here could you please remove white spaces: ": r=", as we have in other log messages (it is easier to grep for an error when it is uniform).